translator/source_code/cotrip_translator.py

```diff
@@ -0,0 +1,421 @@
+def parse_polyline(polylinestring):
+    if not polylinestring or type(polylinestring) != str:
+        return None
+    polyline = polylinestring.split(' ')
+    geometry=[]
+    for i in polyline:
+        coordinates = i.split(',')
+        coords=[]
+        for j in coordinates:
+            coords.append(float(j))
+        geometry.append(coords)
+    return geometry
+
+    
+
+
+
+
+
+# function to get event status
+def get_event_status(start_time_string, end_time_string):
+    start_time = dateutil.parser.parse(start_time_string)
+
+    event_status = "active"
+    if datetime.now(timezone.utc) < start_time:
+        event_status = "planned"  # if < 2 to 3 weeks make it pending instead of planned
+    elif end_time_string:
+        end_time = dateutil.parser.parse(end_time_string)
+
+        if end_time < datetime.now(timezone.utc):
+            event_status = "completed"
+    return event_status
+
+
+
+
+
+
+
+# Parse COtrip alert to WZDx
+def parse_alert(alert, callback_function=None):
+    if not validate_alert(alert):
+        if callback_function:    #Note :a call back fucnction , which will trigger every time the invalid data is given
+            callback_function(alert)
+        return None
+    feature = {}
+    geometry = {}
+    geometry['type'] = "LineString"
+    geometry['coordinates'] = parse_polyline(alert['alert:Polyline'])
+
+    feature['type'] = "Feature"
+    properties = {}
+
+    #### I included a skeleton of the message, fill out all required fields and as many optional fields as you can. Below is a link to the spec page for a road event
+    #### https://github.com/usdot-jpo-ode/jpo-wzdx/blob/master/spec-content/objects/RoadEvent.md
+
+    # road_event_id
+    #### Leave this empty, it will be populated by add_ids
+    properties['road_event_id'] = ''
+
+    # Event Type ['work-zone', 'detour']
+    properties['event_type'] = 'work-zone'
+
+    # data_source_id
+    #### Leave this empty, it will be populated by add_ids
+    properties['data_source_id'] = ''
+
+    # start_date
+    properties['start_date'] = reformat_datetime(alert['alert:ReportedTime'])
+
+    # end_date
+    properties['end_date'] = reformat_datetime(alert.get('alert:ExpectedEndTime', ''))
+
+    # start_date_accuracy
+    properties['start_date_accuracy'] = "estimated"
+
+    # end_date_accuracy
+    properties['end_date_accuracy'] = "estimated"
+
+    # beginning_accuracy
+    properties['beginning_accuracy'] = "estimated"
+
+    # ending_accuracy
+    properties['ending_accuracy'] = "estimated"
+
+    # road_name
+    road_name = alert['alert:RoadName']
+    properties['road_name'] = road_name
+
+    # direction
+    Direction_map = {'North': 'northbound', 'South': 'southbound', 'West': 'westbound', 'East': 'eastbound'}
+ 
+    properties['direction'] = Direction_map.get(alert['alert:Direction'])
+
+    # vehicle impact
+
+    properties['vehicle_impact'] = 'unknown' #get_vehicle_impact(alert['alert:ClosureType'])
```

Comment:
The `get_vehicle_impact` method exists up above, is there a reason we aren't calling it?

Reply:
The data we intended to use to calculate the get vehicle impact is not what we thought and at this point we can not calculate it. Thats why we set it 'unknown'. possibly in future we will use this function.